services/cd-service/pkg/repository/repository.go

```diff
@@ -328,7 +329,32 @@ func (r *repository) drainQueue() []element {
 	}
 }
 
-func (r *repository) ProcessQueueOnce(e element) {
+// DefaultPushUpdate is public for testing reasons only.
+// It returns always nil
+// success is set to true if the push was successful
+func DefaultPushUpdate(branch string, success *bool) git.PushUpdateReferenceCallback {
+	return func(refName string, status string) error {
+		var expectedRefName = fmt.Sprintf("refs/heads/%s", branch)
+		// if we were successful the status is empty and the ref contains our branch:
+		*success = refName == expectedRefName && status == ""
```

Comment:
Uh, this looks hard.

Reply:
I don't know what this means, but I'm also not super happy with this code.
Do you think it won't work in some cases?
I tried it on GitHub and it works fine.

Reply:
After lots of experimentation, I think I'll have to leave it as it is.
We tried running `git init` in a test with a server-side hook in the repo, to reproduce this case where the pushing is not successful. This would avoid this rather ugly callback with the `success` pointer. However, git hooks, incl server-hooks are not supported in this setup, and seem to generally not supported by gitlib2 https://github.com/libgit2/libgit2/blob/main/docs/projects.md?plain=1#L74